### PR TITLE
Change default quantiles to not include endpoints

### LINKED ISSUE #1015: [sdba] Default quantiles are dangerous
As identified in ESPO-R and Spatial Analogs, the default way to choose the quantiles in the quantile mapping adjustments are prompt to generate overfitting of the extremes in the historical period. This is particularly dangerous with precipitation, see figure below as an example.

By default, the `EmpiricalQuantileMapping.train` method takes in a number of quantiles (`nq`, an integer) and calls `quantiles=equally_spaced_nodes(nq, eps=1e-6)`, which generates `nq + 2` quantiles : array of `nq` equally spaced numbers ranging from `1 / (nq + 1)` to `nq / (nq + 1)` PLUS `0.000001` and `0.999999`.  

In many cases, the last two additions are essentially the minimum and the maximum and we are effectively clipping the adjusted data on the historical period to the min and max of the reference. As soon as a simulation has some values _outside_ this historical period that are larger (or lower) then its own min and max _within_ the historical period, those peaks appear completely out-of-place compared to a flattened historical scenario. See figure below.

image
Showing the annual maximal precipitation for a single point, 24 realizations, as adjusted with `DetrendedQuantileMapping`, with 50 quantiles and a 30-yr rolling-mean detrending.

The `DetrendedQuantileMapping` and `QuantileDeltaMapping` objects are also affected by this issue because they rely on EQM's train.

My suggestion is to default to `esp=None`.

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1015
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.
- [ ] `bumpversion patch` has been called on this branch
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* Change the default behaviour of QM algos to not include endpoints in the quantile vector.
The documentation was changed accordingly and a warning was added in the docstring of `sdba.utils.equally_spaced_nodes`.

### Does this PR introduce a breaking change?
Yes. The way to get back to the previous behaviour is documented in the history entry.

### Other information:
Should I add a warning?

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,22 @@
 History
 =======
 
+0.34.0 (unreleased)
+-------------------
+Contributors to this version: Pascal Bourgault (:user:`aulemahal`).
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+* Quantile mapping adjustment objects (EQM, DQM and QDM) and ``sdba.utils.equally_spaced_nodes`` will not add additionnal endpoints to the quantile range. (:issue:`1015`, :pull:`1016`). To retrieve the same functionality as before use.
```
Comment (huard) [inline HISTORY.rst:11]:
```suggestion
* Quantile mapping adjustment objects (EQM, DQM and QDM) and ``sdba.utils.equally_spaced_nodes`` will not add additionnal endpoints to the quantile range. (:issue:`1015`, :pull:`1016`). To retrieve the same functionality as before use:
```

File: xclim/sdba/utils.py [hunk 2]
```diff
@@ -236,20 +236,24 @@ def broadcast(
     return grouped
 
 
-def equally_spaced_nodes(n: int, eps: Union[float, None] = 1e-4) -> np.array:
-    """Return nodes with `n` equally spaced points within [0, 1] plus two end-points.
+def equally_spaced_nodes(n: int, eps: Optional[float] = None) -> np.array:
+    """Return nodes with `n` equally spaced points within [0, 1], optionally adding two end-points.
+
+    WARNING: Passing a small `eps` will effectively clip the scenario to the bounds of the reference
```
Comment (huard) [inline xclim/sdba/utils.py:242]:
```suggestion
    .. warning:: Passing a small `eps` will effectively clip the scenario to the bounds of the reference
```

File: HISTORY.rst (hunk 1 above)
Comment (huard) [reply HISTORY.rst:11]: Add a sentence explaining the rationale for this change.

**Review by huard — APPROVED** [review]

File: xclim/sdba/utils.py [hunk 3]
```diff
@@ -239,7 +239,7 @@ def broadcast(
 def equally_spaced_nodes(n: int, eps: Optional[float] = None) -> np.array:
     """Return nodes with `n` equally spaced points within [0, 1], optionally adding two end-points.
 
-    WARNING: Passing a small `eps` will effectively clip the scenario to the bounds of the reference
+    .. warning:: Passing a small `eps` will effectively clip the scenario to the bounds of the reference
```
Comment (Zeitsperre) [inline xclim/sdba/utils.py]: Why not use the:
```
Warnings
--------
```
RsT convention?

Comment (aulemahal) [reply xclim/sdba/utils.py]: I thought the "Warnings" section, as you suggest, was for listing what python warnings could be raised and why, while this rst block will show up in the main docstring as a not to the user.

After reading https://numpydoc.readthedocs.io/en/latest/format.html#warnings, I see that I am wrong.

**Pull request merged**